examples/twittermap/web/app/controllers/TwitterMapApplication.scala

```diff
@@ -197,6 +209,49 @@ object TwitterMapApplication {
     }
   }
 
+  /** Find cities' population whose cityID are in cityIDs.
+    *
+    * @param cityIds List of cities in current boundary in String
+    * @param citiesPopulation List of all cities' population data
+    * @return List of cities population which centroids is in current boundary
+    */
+  def findCityPop(cityIds: String, citiesPopulation: List[JsValue]): JsArray = {
+    val cityIdsList : List[Int] = cityIds.split(",").map(_.toInt).toList
+    val incCityIdsList = cityIdsList.sortWith(_<_)
+    sortedListBinarySearch(citiesPopulation, 0, 29833, incCityIdsList, 0, Json.arr())
+  }
+
+  /**
+    * Find the population data data for each geoId in from the sorted geoIds list.
+    *
+    * @param sortedCityIds Sorted list of cities in current boundary
+    * @param citiesPopulation List of all cities' population data
+    * @return List of cities in sortedCityIds' population data
+    */
+  def sortedListBinarySearch(citiesPopulation: List[JsValue], startIndex: Int, endIndex: Int,
```

Comment:
So we need the `allCityPopulation.json` sorted by cityID in ascending order, correct?

Reply:
Yes.